### PR TITLE
Test/unit tests student

### PR DESCRIPTION
**Pull request by tiago-falves**
Created Unit tests for student
Corrected many errors in class student

### CODE REVIEW

File: src/student.py [hunk 1]
```diff
@@ -7,22 +7,65 @@ def __init__(self, student_id, subjects_and_classes, subject_targets, subject_gi
         self.buddies = buddies   #Dictionary where for each subject (key) there is a list of ups ordered by their priority
     
     def add_subject_and_class(self, subject_name, class_number):
-        self.subjects_and_classes[subject_name] = class_number
+        if len(self.subjects_and_classes) < 7:
```
Comment (FilipePintoReis) [inline src/student.py:10]: @tiago-falves where does this 7 come from?

**Review by FilipePintoReis — DISMISSED** [review]

File: src/student.py (hunk 1 above)
Comment (tiago-falves) [reply src/student.py:10]: It is the maximum amout of classes a student can do in a semester!

**Review by Jumaruba — DISMISSED** [review]
Just for good code practice and future utility, it would be nice having the functions returning 0 upon success and 1 otherwise

**Comment by FilipePintoReis** [comment]
> Just for good code practice and future utility, it would be nice having the functions returning 0 upon success and 1 otherwise

When you request changes you must mention the line

File: src/student.py [hunk 2]
```diff
@@ -7,22 +7,65 @@ def __init__(self, student_id, subjects_and_classes, subject_targets, subject_gi
...
+            self.subjects_and_classes[subject_name] = class_number
 
     def add_subject_target(self, subject_name, class_number):
-        self.subject_targets[subject_name].append(class_number)
+        if subject_name in self.subjects_and_classes:
+            if subject_name not in self.subject_targets:
+                self.subject_targets[subject_name] = [class_number]
+            else:
+                self.subject_targets[subject_name].append(class_number)
+
```
Comment (Jumaruba) [inline src/student.py]: Just for good code practice and future utility, it would be nice having the functions returning 0 upon success and 1 otherwise.

**Review by Jumaruba — CHANGES_REQUESTED** [review]

File: src/student.py (hunk 2 above)
Comment (tiago-falves) [reply src/student.py]: Already done!

**Review by Jumaruba — APPROVED** [review]

**Review by FilipePintoReis — APPROVED** [review]

**Pull request merged**